src/librustc/ty/mod.rs

```diff
@@ -2558,6 +2560,22 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
             Err(self.sess.cstore.crate_name(impl_did.krate))
         }
     }
+
+    pub fn adjust(self, name: Name, scope: DefId, block: NodeId) -> (Ident, DefId) {
+        self.adjust_ident(name.to_ident(), scope, block)
+    }
+
+    pub fn adjust_ident(self, mut ident: Ident, scope: DefId, block: NodeId) -> (Ident, DefId) {
```

Comment:
We adjust immediately before resolving a name in a scope. Ideally we'd only call the `adjust` functions in a generic "resolve name in scope" function, but that would require more refactoring to unify the various environments in which we resolve names today.